zfit/models/physics.py

```diff
@@ -888,3 +888,295 @@ class GeneralizedGaussExpTailPDFRepr(BasePDFRepr):
+    q4 = tf.constant([1.0, 106.8615961, 337.6496214, 2016.712389, 1597.063511])
+
+    p5 = tf.constant([1.003675074, 167.5702434, 4789.711289, 21217.86767, -22324.94910])
+    q5 = tf.constant([1.0, 156.9424537, 3745.310488, 9834.698876, 66924.28357])
+
+    p6 = tf.constant([1.000827619, 664.9143136, 62972.92665, 475554.6998, -5743609.109])
+    q6 = tf.constant([1.0, 651.4101098, 56974.73333, 165917.4725, -2815759.939])
+
+    a1 = tf.constant([0.04166666667, -0.01996527778, 0.02709538966])
+    a2 = tf.constant([-1.845568670, -4.284640743])
+
+    if sigma <= 0:
+        return 0
+
+    v = (x - mu) / sigma
+    denlan = 0.0
+
+    # the thresholds set for the different cases
+    THRESHOLDS_PDF = tf.constant([-5.5, -1.0, 1.0, 5.0, 12.0, 50.0, 300.0])
+
+    # the different function definitions for each case : how to process v depending on its value
+    def case_less_than_minus5_5_PDF(v):
+        u = znp.exp(v + 1.0)
+        ue = znp.exp(-1 / u)
+        us = znp.sqrt(u)
+
+        return tf.case(
+            [tf.less(u, tf.constant(1e-10))],
+            lambda: tf.constant(0),
+            lambda: 0.3989422803 * (ue / us) * (1 + (a1[0] + (a1[1] + a1[2] * u) * u) * u),
+        )
+
+    def case_less_than_minus1_PDF(v):
+        u = znp.exp(-v - 1)
+        return (
+            znp.exp(-u)
+            * znp.sqrt(u)
+            * (p1[0] + (p1[1] + (p1[2] + (p1[3] + p1[4] * v) * v) * v) * v)
+            / (q1[0] + (q1[1] + (q1[2] + (q1[3] + q1[4] * v) * v) * v) * v)
+        )
+
+    def case_less_than_1_PDF(v):
+        return (p2[0] + (p2[1] + (p2[2] + (p2[3] + p2[4] * v) * v) * v) * v) / (
+            q2[0] + (q2[1] + (q2[2] + (q2[3] + q2[4] * v) * v) * v) * v
+        )
+
+    def case_less_than_5_PDF(v):
+        return (p3[0] + (p3[1] + (p3[2] + (p3[3] + p3[4] * v) * v) * v) * v) / (
+            q3[0] + (q3[1] + (q3[2] + (q3[3] + q3[4] * v) * v) * v) * v
+        )
+
+    def case_less_than_12_PDF(v):
+        u = 1 / v
+        return (
+            u
+            * u
+            * (p4[0] + (p4[1] + (p4[2] + (p4[3] + p4[4] * u) * u) * u) * u)
+            / (q4[0] + (q4[1] + (q4[2] + (q4[3] + q4[4] * u) * u) * u) * u)
+        )
+
+    def case_less_than_50_PDF(v):
+        u = 1 / v
+        return (
+            u
+            * u
+            * (p5[0] + (p5[1] + (p5[2] + (p5[3] + p5[4] * u) * u) * u) * u)
+            / (q5[0] + (q5[1] + (q5[2] + (q5[3] + q5[4] * u) * u) * u) * u)
+        )
+
+    def case_less_than_300_PDF(v):
+        u = 1 / v
+        return (
+            u
+            * u
+            * (p6[0] + (p6[1] + (p6[2] + (p6[3] + p6[4] * u) * u) * u) * u)
+            / (q6[0] + (q6[1] + (q6[2] + (q6[3] + q6[4] * u) * u) * u) * u)
+        )
+
+    def default_case(v):
+        u = 1 / (v - v * znp.log(v) / (v + 1))
+        return u * u * (1 + (a2[0] + a2[1] * u) * u)
+
+    # this is the equivalent of a long line of if / elif conditions.
+    # exlusive != False
+    denlan = tf.case(
```

Comment:
On all this `tf.case`: they're not a branching on a _value_, but rather on a whole branch. That means it's not properly vectorized and the way to go is, afaiu, rather a vectorizez `znp.where` instead.

Technically, there is no real way around it, afaiu vectorized computation